hoomd/mpcd/test/sphere_geometry_rejection_filler_test.cc

```diff
@@ -0,0 +1,177 @@
+        unsigned int N_out(0);
+        for (unsigned int i=pdata->getN(); i < pdata->getN() + pdata->getNVirtual(); ++i)
+            {
+            // tag should equal index on one rank with one filler
+            UP_ASSERT_EQUAL(h_tag.data[i], i);
+            // type should be set
+            UP_ASSERT_EQUAL(__scalar_as_int(h_pos.data[i].w), 1);
+
+            Scalar3 pos = make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > r*r)
+                ++N_out;
+            }
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        UP_ASSERT_GREATER(N_out, Nfill_0);
+        }
+
+    /*
+     * Test the average properties of the virtual particles.
+     */
+    // initialize variables for storing avg data
+    Scalar N_avg(0);
+    Scalar3 vel_avg_net = make_scalar3(0,0,0);
+    Scalar T_avg(0);
+    // repeat filling 100 times
+    unsigned int itr(1000);
+    for (unsigned int t=0; t<itr; ++t)
+        {
+        pdata->removeVirtualParticles();
+        filler->fill(2+t);
+
+        ArrayHandle<Scalar4> h_pos(pdata->getPositions(), access_location::host, access_mode::read);
+        ArrayHandle<Scalar4> h_vel(pdata->getVelocities(), access_location::host, access_mode::read);
+
+        // local variables
+        unsigned int N_out(0);
+        Scalar temp(0);
+        Scalar3 vel_avg = make_scalar3(0,0,0);
+
+        for (unsigned int i=pdata->getN(); i < pdata->getN() + pdata->getNVirtual(); ++i)
+            {
+            const Scalar3 pos = make_scalar3(h_pos.data[i].x, h_pos.data[i].y, h_pos.data[i].z);
+            const Scalar3 vel = make_scalar3(h_vel.data[i].x, h_vel.data[i].y, h_vel.data[i].z);
+            const Scalar r2 = dot(pos, pos);
+            if (r2 > r*r)
+                ++N_out;
+            temp += dot(vel, vel);
+            vel_avg += vel;
+            }
+
+        temp /= (3*(N_out-1));
+        vel_avg_net += vel_avg/N_out;
+        // Check whether all virtual particles are outside the sphere
+        UP_ASSERT_EQUAL(N_out, pdata->getNVirtual());
+        N_avg += N_out;
+        T_avg += temp;
+
+        }
+    N_avg /= itr;
+    T_avg /= itr;
+    vel_avg_net /= itr;
+
+    /*
+    * Expected number of virtual particles = int( density * volume outside sphere )
+    * volume outside sphere = sim-box volume - sphere volume
+    * N_exptd = int(density*(L^3 - 4*pi*r^3/3))
+    *         = 14952
+    */
+    UP_ASSERT_CLOSE(N_avg, 14952.0, 2);
+
+    CHECK_SMALL(vel_avg_net.x, tol_small);
+    CHECK_SMALL(vel_avg_net.y, tol_small);
+    CHECK_SMALL(vel_avg_net.z, tol_small);
+    CHECK_CLOSE(T_avg, 1.5, tol_small);
+    }
+
+UP_TEST( sphere_rejection_fill_basic )
+    {
+    sphere_rejection_fill_basic_test<mpcd::RejectionVirtualParticleFiller<mpcd::detail::SphereGeometry>>(std::make_shared<ExecutionConfiguration>(ExecutionConfiguration::CPU));
+    }
+#ifdef ENABLE_CUDA
```

Comment:
Should this be `ENABLE_HIP` instead? You can verify if this test is actually running by adding a print statement in this test case.